packages/typespec-client-generator-core/src/types.ts

```diff
@@ -837,11 +837,30 @@ export function getSdkModelWithDiagnostics(
     }
 
     // properties should be generated first since base model's discriminator handling is depend on derived model's properties
-    if (operation && isMultipartOperation(context, operation)) {
-      const body = getHttpOperationWithCache(context, operation).parameters.body;
-      if (body && getHttpBodyType(body) === type) {
-        // handle multipart body model properties
-        diagnostics.pipe(addMultipartPropertiesToModelType(context, sdkType, operation));
+    if (operation) {
+      const requestBody = getHttpOperationWithCache(context, operation).parameters.body;
+      const multipartResponseBodies = getHttpOperationWithCache(context, operation)
+        .responses.map((response) => response.responses.map((r) => r.body))
+        .flatMap((x) => x)
+        .filter((x) => x?.bodyKind === "multipart");
+      if (
+        requestBody &&
+        requestBody.bodyKind === "multipart" &&
+        getHttpBodyType(requestBody) === type
+      ) {
+        // handle multipart request body model properties
+        diagnostics.pipe(
+          addMultipartPropertiesToModelType(context, sdkType, requestBody, operation),
+        );
+      } else if (multipartResponseBodies.length > 0) {
+        // handle multipart response body model properties
```

Comment:
I am not sure why this 2 block is `if .. else ..`, as one handle request, one handle response?

Reply:
If a model is used both in request and response, we could do the properties handling only once.